features/bootstrap/JsonApiContext.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\DummyFriend;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\RelatedDummy;
+use Behat\Behat\Context\Context;
+use Behat\Behat\Context\Environment\InitializedContextEnvironment;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behatch\Context\RestContext;
+use Behatch\Json\Json;
+use Behatch\Json\JsonInspector;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+final class JsonApiContext implements Context
+{
+    private $restContext;
+
+    private $inspector;
+
+    private $doctrine;
+
+    /**
+     * @var \Doctrine\Common\Persistence\ObjectManager
+     */
+    private $manager;
+
+    public function __construct(ManagerRegistry $doctrine)
+    {
+        $this->doctrine = $doctrine;
+        $this->manager = $doctrine->getManager();
+    }
+
+    /**
+     * Gives access to the Behatch context.
+     *
+     * @param BeforeScenarioScope $scope
+     *
+     * @BeforeScenario
+     */
+    public function gatherContexts(BeforeScenarioScope $scope)
+    {
+        /** @var InitializedContextEnvironment $environment */
+        $environment = $scope->getEnvironment();
+
+        $this->restContext = $environment->getContext(RestContext::class);
+
+        $this->inspector = new JsonInspector('javascript');
+    }
+
+    /**
+     * @Then I save the response
+     */
+    public function iSaveTheResponse()
+    {
+        $content = $this->getContent();
+
+        if (null === ($decoded = json_decode($content))) {
+            throw new \RuntimeException('JSON response seems to be invalid');
+        }
+
+        $fileName = __DIR__.'/response.json';
+
+        file_put_contents($fileName, $content);
+
+        return $fileName;
+    }
+
+    /**
+     * @Then I validate it with jsonapi-validator
+     */
+    public function iValidateItWithJsonapiValidator()
+    {
+        $fileName = $this->iSaveTheResponse();
+
+        $validationResponse = exec(sprintf('cd %s && jsonapi-validator -f response.json', __DIR__));
```

Comment:
I repost my comment from https://github.com/api-platform/core/pull/785#discussion_r84666498 here, I'm really not a fan of this.